### PR TITLE
update release doc

### PR DESCRIPTION
**Pull request by gabrielwen**
Adding instructions for joining release team and updating application version.

---
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/kubeflow/3968)

### CODE REVIEW

File: docs_dev/releasing.md [hunk 1]
```diff
@@ -342,6 +347,10 @@ You can create a release branch via the GitHub UI.
 
 A release candidate is a tag of the form `v${MAJOR}.${MINOR}.${PATCHLEVEL}-rc.${N}`, where `N` is a small integer, and a release candidate tag always points to a commit on the corresponding minor release branch.  Push this tag to GitHub and announce a release vote on kubeflow-discuss.
 
+### Update Version Shown on Central Dashboard
+
+Update release tag to central dashboard: [link](https://github.com/kubeflow/manifests/blob/eab125dd42c08b0e2495b7b2d7a7010ac64d3774/application/application/overlays/application/application.yaml#L16)
```
Comment (jlewi) [inline docs_dev/releasing.md]: Can you clarify this section; we aren't updating the central dashboard.
We are updating the version specified in the Kubeflow application resource 
(the fact that central dashboard is displaying that is a secondary effect).

File: docs_dev/releasing.md [hunk 2]
```diff
@@ -156,6 +158,9 @@ Use [this script](https://github.com/jlewi/kubeflow-dev/blob/master/create_conte
 create_context.sh $(kubectl config current-context) kubeflow-releasing
```
Comment (jlewi) [inline docs_dev/releasing.md]: Can you delete the section about building TFServing images please; that step is no longer needed.

**Comment by jlewi** [comment]
@gabrielwen Could you do a pass through the release doc please and update any sections that no longer make sense; there is a lot of outdated information.

I think it would be great to talk about the following

1. How to prepare individual applications for release

   * e.g. for TFJob and PYtorch; cut release branches for those applications
   * Update kubeflow/manifests master to use them

1. How to cut a release branch of kubeflow/manifests and prepare it for relesae

    * e.g. update the Kubeflow application resource version to the upcoming major.minor release

1. How to cut a kfctl release
    
    * e.g. cut the branch; update releases etc.

1. How to update the KFDef specs on the release branch

       * The repos section should be pinned to specific commits on the release branch


We can also delete a bunch of outdated references to ksonnet
e.g. update or delete all references to ksonnet; e.g. the section **Updating ksonnet prototypes with docker image** and the section "Updating the ksonnet configs for master"

**Comment by jlewi** [comment]
We should also add some steps about looking at the test grid to ensure tests are passing.

**Comment by jlewi** [comment]
@gabrielwen Think we can wrap this up this week?

**Comment by gabrielwen** [comment]
@jlewi should be ready, could you take a look?  thanks.

File: docs_dev/releasing.md [hunk 3]
```diff
@@ -181,78 +179,6 @@ Identify the [release](https://github.com/kubeflow/pytorch-operator/releases) of
...
-ks param set --env=${ENV} workflows namespace kubeflow-releasing
-ks param set --env=${ENV} workflows name "${USER}-${JOB_NAME}-${VERSION_TAG}"
-ks param set --env=${ENV} workflows prow_env "JOB_NAME=${JOB_NAME},JOB_TYPE=${JOB_TYPE},PULL_BASE_SHA=${PULL_BASE_SHA},REPO_NAME=${REPO_NAME},REPO_OWNER=${REPO_OWNER},BUILD_NUMBER=${BUILD_NUMBER}"
-ks param set --env=${ENV} workflows versionTag ${VERSION_TAG}
-ks apply ${ENV} -c workflows
-```
-
-## Build the Jupyter Images
-
-We build the Jupyter using our E2E argo workflows.
```
Comment (jlewi) [inline docs_dev/releasing.md]: How do we release jupyter images? Its possible the postsubmits are automatically pushing to the public repo but I'm not sure.

File: docs_dev/releasing.md [hunk 4]
```diff
@@ -150,12 +151,9 @@ gcloud config set project kubeflow-releasing
 gcloud container clusters get-credentials kubeflow-releasing --zone us-central1-a --project kubeflow-releasing
 ```
 
-Use [this script](https://github.com/jlewi/kubeflow-dev/blob/master/create_context.sh) to set up your context properly:
-
-```
-create_context.sh $(kubectl config current-context) kubeflow-releasing
-```
+## Authorization to Publish a Release
```
Comment (jlewi) [inline docs_dev/releasing.md]: How about creating a section pre-requisites for creating a release

File: docs_dev/releasing.md [hunk 5]
```diff
@@ -150,12 +151,9 @@ gcloud config set project kubeflow-releasing
...
 
+Need to join [release team](https://github.com/kubeflow/internal-acls/blob/1234654eb219e2c06ed5fdc2c4e662a02fccc740/github-orgs/kubeflow/org.yaml#L388) before you can publish a release.
```
Comment (jlewi) [inline docs_dev/releasing.md]: How about adding a section 

Getting ready to cut a release 

and adding the following

1. Create a P0 bug of kind process to track the major release.
1. Announce the release and link to the issue in the #release channel in kubeflow.slack.com

**Comment by jlewi** [comment]
@gabrielwen see my previous [comment](https://github.com/kubeflow/kubeflow/pull/3968#issuecomment-523647132) it looks like we are still missing a bunch of information.

Here's my meta comment. Can we try to define a clear process that would be easy for someone to follow?

Lets add a section about releasing all the individual components 

In this section lets create a list of all the components that should be updated as part of each release

* TFJob
* PyTorch
* JupyterWebapp
* CentralDashboard
* Katib
* ProfileController
* NotebookController

The instructions should say something like the following

1. Create release branch for all of the above
1. Create a Kubernetes periodic prow job for the release branch
1. Update the prow_config.yaml on the release branch if necessary 
    * i.e. kfctl v0.Y-branch should pull in v0.Y branch of kubeflow-manifests
1. Check that postsubmits on the branch are green
1. Build and publish docker images to gcr.io/kubeflow-images-public for all of the above
1. Update kustomize manifest on kubeflow/manifest to reference the above


Once all of the application manifests on kubeflow/master have been updated we are ready to cut a release branch

Once we cut the release branch of kubeflow/manifests lets list out all the steps that need to be performed

1. Create a periodic prow job
1. Update prow_config to use the v0.Y branch of kfctl
1. Update application version in all applications

File: docs_dev/releasing.md (hunk 3 above)
Comment (gabrielwen) [reply docs_dev/releasing.md]: yes, it does.  we confirmed:
https://github.com/kubeflow/manifests/commit/723f310a60c9765bcadc2ed75053d5819f429c60

**Comment by jlewi** [comment]
@gabrielwen let me know when this is ready for another look?

**Comment by gabrielwen** [comment]
working on this now.

File: docs_dev/releasing.md (hunk 4 above)
Comment (gabrielwen) [reply docs_dev/releasing.md]: done

File: docs_dev/releasing.md (hunk 5 above)
Comment (gabrielwen) [reply docs_dev/releasing.md]: done

**Comment by gabrielwen** [comment]
@jlewi updated, could you also take a look?  I'll also take second pass tonight.

**Comment by jlewi** [comment]
This is great; thanks @gabrielwen 

/lgtm
/approve

**Pull request merged**